PROTO_tests/tests/numeric_test.py

```diff
@@ -197,22 +204,249 @@ def test_cumprod(self, num_type):
+    def test_arccos(self, num_type):
+        if num_type == ak.uint64:
+            na = np.arange(0, 2).astype(num_type)
+        elif num_type == ak.float64:
+            na = np.linspace(-1, 1)
+        else:
+            na = np.arange(-1, 2).astype(num_type)
+        pda = ak.array(na, dtype=num_type)
+
+        assert np.allclose(np.arccos(na), ak.arccos(pda).to_ndarray(), equal_nan=True)
+        with pytest.raises(TypeError):
+            ak.arccos(np.array([range(0, 10)]).astype(num_type))
+
+        # Edge case: Infinities and NaNs
+        if num_type == ak.float64:
+            na = np.array([np.inf, np.nan, -np.inf])
+            pda = ak.array(na)
+            assert np.allclose(np.arccos(na), ak.arccos(pda).to_ndarray(), equal_nan=True)
+
+    @pytest.mark.parametrize("num_type", NO_BOOL)
+    def test_arctan(self, num_type):
+        na = NP_TRIG_ARRAYS[num_type]
+        pda = ak.array(na, dtype=num_type)
+
+        assert np.allclose(np.arctan(na), ak.arctan(pda).to_ndarray(), equal_nan=True)
+        with pytest.raises(TypeError):
+            ak.arctan(np.array([range(0, 10)]).astype(num_type))
+
+        # Edge case: infinities and NaNs
+        if num_type == ak.float64:
+            na = np.array([np.inf, np.nan, -np.inf])
+            pda = ak.array(na)
+            assert np.allclose(np.arctan(na), ak.arctan(pda).to_ndarray(), equal_nan=True)
+
+    @pytest.mark.parametrize("num_type", NO_BOOL)
+    @pytest.mark.parametrize("denom_type", NO_BOOL)
+    def test_arctan2(self, num_type, denom_type):
+        na_num = NP_TRIG_ARRAYS[num_type]
+        na_denom = np.arange(0, 10).astype(denom_type)
+
+        pda_num = ak.array(na_num, dtype=num_type)
+        pda_denom = ak.array(na_denom, dtype=denom_type)
+
+        num = np.array([1]).astype(num_type)
+        denom = np.array([5]).astype(denom_type)
+
+        assert np.allclose(
+            np.arctan2(na_num, na_denom), ak.arctan2(pda_num, pda_denom).to_ndarray(), equal_nan=True
+        )
+        assert np.allclose(
+            np.arctan2(num[0], na_denom), ak.arctan2(num[0], pda_denom).to_ndarray(), equal_nan=True
+        )
+        assert np.allclose(
+            np.arctan2(na_num, denom[0]), ak.arctan2(pda_num, denom[0]).to_ndarray(), equal_nan=True
+        )
+
+        with pytest.raises(TypeError):
+            ak.arctan2(
+                np.array([range(0, 10)]).astype(num_type), np.array([range(10, 20)]).astype(num_type)
+            )
+            ak.arctan2(num[0], np.array([range(10, 20)]).astype(num_type))
+            ak.arctan2(np.array([range(0, 10)]).astype(num_type), denom[0])
+
+        # Edge case: Infinities
+        na_inf = np.array([np.inf, np.nan, -np.inf])
+        pda_inf = ak.array(na_inf)
+        na_fin = np.array([0, 5, 10]).astype(denom_type)
+        pda_fin = ak.array(na_fin)
+        assert np.allclose(
+            np.arctan2(na_inf, na_fin), ak.arctan2(pda_inf, pda_fin).to_ndarray(), equal_nan=True
+        )
```

Comment:
I think this should be handled by the above once we make the `np.random.permutation` change

Reply:
You are correct that these are now handled, but the next two test are for the vector-scalar and scalar-vector cases. So I will remove the repeated testing.

Reply:
Updated